proxy/auth_test.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Stefan Prodan 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package proxy
+
+import (
+	"net/http"
+	"testing"
+
+	"strings"
+
+	"github.com/openfaas/faas-cli/config"
+)
+
+func Test_AuthorizationHeader(t *testing.T) {
+	//setup store
+	config.DefaultFile = "authtest1.yml"
+	gatewayURL := strings.TrimRight("http://openfaas.test/", "/")
+	config.UpdateAuthConfig(gatewayURL, "Aladdin", "open sesame")
```

Comment:
Have we tested for empty string and edge-cases?

Reply:
The URL empty string validation is done here https://github.com/openfaas/faas-cli/pull/182/files/dc06babac728c33fb128bba45766f229c134a991#diff-4d2d07a462d7a1dda6e42b8c443d6d77R74 and the validity of the URL is tested by calling the gateway `/list` before the URL get's saved in the config.

Reply:
My question was - have we got a unit test?

Reply:
I'm also unsure we should be using username password. My original PR only did it because it incorrectly used cli-arguments. We are storing base64, why not add the requisite header?

Reply:
testes added for empty string and invalid URLs in https://github.com/openfaas/faas-cli/pull/182/commits/35df33c431d82e5137dfd0928c48940d185d2170